[torch][repeat_interleave] Fix ambigious function call

serhaty opened the pull request:
Summary:
recently added new parameter to the function with PR: https://github.com/pytorch/pytorch/pull/58417

However, this introduced ambiguity when making call below:
  some_tensor.repeat_interleave(some_integer_value)

Making it optional to avoid the issue.

Differential Revision: D28653820

ezyang reviewed (approved):
This is BC breaking you probably need to update the BC test too

Commit:
[torch][repeat_interleave] Fix ambigious function call (#58881)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/58881

recently added new parameter to the function with PR: https://github.com/pytorch/pytorch/pull/58417

However, this introduced ambiguity when making call below:
  some_tensor.repeat_interleave(some_integer_value)

Making it optional to avoid the issue.

Reviewed By: ezyang

Differential Revision: D28653820

fbshipit-source-id: 0566490fe3414d2c5ea14f5b82b02f10708cdf82
